src/provider/mod.rs

```diff
@@ -983,20 +1002,29 @@ async fn create_collection_inner(
     let mut blobs = Vec::with_capacity(data_sources.len());
     let mut total_blobs_size: u64 = 0;
 
+    data_sources.sort();
+
     // compute outboards in parallel, using tokio's blocking thread pool
-    let outboards = data_sources.into_iter().map(|data| {
-        let (path, name) = match data {
-            DataSource::File(path) => (path, None),
-            DataSource::NamedFile { path, name } => (path, Some(name)),
-        };
-        tokio::task::spawn_blocking(move || compute_outboard(path, name))
-    });
+    let outboards = stream::iter(data_sources)
+        .map(|data| {
+            let (path, name) = match data {
+                DataSource::File(path) => (path, None),
+                DataSource::NamedFile { path, name } => (path, Some(name)),
+            };
+            tokio::task::spawn_blocking(move || compute_outboard(path, name))
+        })
+        // allow at most num_cpus tasks at a time, otherwise we might get too many open files
+        .buffer_unordered(num_cpus::get());
```

Comment:
Yes, this assumes that the task is fully CPU bound, which is probably not the case.

But for all systems where num_cpus < max open files, this conveniently limits the number of open files, so I guess it will have to do for now. I will add a todo.